source/s3_auto_ranged_put.c

```diff
@@ -1412,27 +1171,22 @@ static int s_s3_review_multipart_upload(struct aws_s3_request *request) {
     }
 }
 
-/* Finish async preparation of a CompleteMultipartUpload request */
-static void s_s3_prepare_complete_multipart_upload_finish(
-    struct aws_s3_prepare_complete_multipart_upload_job *complete_mpu_prep,
-    int error_code) {
-
-    struct aws_s3_request *request = complete_mpu_prep->request;
+/* Prepare a CompleteMultipartUpload request. */
+static struct aws_future_http_message *s_s3_prepare_complete_multipart_upload(struct aws_s3_request *request) {
     struct aws_s3_meta_request *meta_request = request->meta_request;
     struct aws_s3_auto_ranged_put *auto_ranged_put = meta_request->impl;
+    struct aws_allocator *allocator = request->allocator;
 
-    if (error_code != AWS_ERROR_SUCCESS) {
-        aws_future_http_message_set_error(complete_mpu_prep->on_complete, error_code);
-        goto on_done;
-    }
+    struct aws_future_http_message *message_future = aws_future_http_message_new(allocator);
 
     AWS_FATAL_ASSERT(auto_ranged_put->upload_id);
+    int error_code = AWS_OP_SUCCESS;
```

Comment:
don't need this anymore, since we're not pulling an error_code out of a future anymore
```suggestion
```